classes/GunImage.py

```diff
@@ -0,0 +1,147 @@
+        :param guns_data: current iteration of the gun data
+        :param gun_type: gun type to filter on
+        :param manufacturer: manu to filter on
+        :param name: name to search for
+        :return: recursive call for more filtering or successfully filtered set
+        """
+        filtered_guns = []
+        if name is not None:
+            for item in guns_data:
+                if item.get("name").lower() == name.lower():
+                    filtered_guns.append(item)
+                    print(item)
+            return self.filter_guns_data(guns_data=filtered_guns, gun_type=gun_type, manufacturer=manufacturer)
+        
+        if manufacturer is not None:
+            for item in guns_data:
+                if item.get("manufacturer").lower() == manufacturer.lower():
+                    filtered_guns.append(item)
+            return self.filter_guns_data(guns_data=filtered_guns, gun_type=gun_type)
+        
+        if gun_type is not None:
+            for item in guns_data:
+                if item.get("type").lower() == gun_type.lower():
+                    filtered_guns.append(item)
+            return filtered_guns
+        
+        return guns_data
+
+    def sample_gun_image(self, gun_type=None, manufacturer=None):
+        """
+        Handles sampling and downloading a relevant gun image from the games for gun card display use
+        :param gun_type: type to filter on
+        :param manufacturer: manu/guild
+        """
+        # Error catch if empty
+        if gun_type is None and manufacturer is None:
+            raise FileNotFoundError("No type or manufacturer given!")
+
+        # Convert guild name
+        manufacturer = self.manu_conversion(manufacturer)
+
+        # Convert gun type
+        gun_type = self.type_conversion(gun_type)
+
+        # For each given gun type, get its filtered data and concatenate
+        temp_data = []
+        for entry in gun_type:
+            temp_data.append(
+                self.filter_guns_data(self.guns_data, entry, manufacturer)
+            )
+
+        gun_data = []
+        for temp in temp_data:
+            gun_data.extend(temp)
+
+        # Get a sample and its url link
+        print(gun_data)
+        sample = random.sample(gun_data, 1)[0]['image_link']
+
+        # Download image and convert to a temporary local PDF
+        # TODO - swap out with legit urls when it gets updated
+        url = [
+            "https://global-uploads.webflow.com/5ff36780a1084987868ce198/618fd0cdd6736d4a1bbf5fbe_9-Volt%20(SMG-BL3).png",
+            "https://global-uploads.webflow.com/5ff36780a1084987868ce198/618fdc07ad989ce7693372d0_Narp%20(Sniper-BL3).png"
+        ]
+        url = random.sample(url, 1)[0]
+
+        response = requests.get(url, stream=True)
+        img = Image.open(response.raw)
+        img.save('temporary_gun_image.png')
+
+    def manu_conversion(self, guild_name):
```

Comment:
Using a dict for this is more efficient than direct comparison:

```
manu_map = {
  "alas!": "Atlas"
}

return manu_map.get(guild_name, None)
```

Reply:
Ahhh good catch. Fever dream late night pumping lmao